components/openzwave-adapter/src/lib.rs

```diff
@@ -454,7 +486,20 @@ impl OpenzwaveAdapter {
                             }
                         }
                     }
-                    ZWaveNotification::ValueRemoved(_value)         => {}
+                    ZWaveNotification::ValueRemoved(vid)            => {
+                        if let Some(getter_id) = getter_map.find_taxo_id_from_ozw(&vid) {
+                            box_manager.remove_channel(&getter_id).unwrap_or_else(|e| {
+                                error!("Unable to remove getter_id {}: {}", getter_id, e);
+                            });
+                            getter_map.remove_by_ozw(&vid);
+                        }
+                        if let Some(setter_id) = setter_map.find_taxo_id_from_ozw(&vid) {
+                            box_manager.remove_channel(&setter_id).unwrap_or_else(|e| {
+                                error!("Unable to remove setter_id {}: {}", setter_id, e);
+                            });
+                            setter_map.remove_by_ozw(&vid);
+                        }
```

Comment:
same here, you can use `remove_by_ozw` directly instead of `find_taxo_id_from_ozw` if it returns the removed item. And I think we'd also be threadsafe thanks to the guard in remove_by_ozw.